cyclonedx/model/component.py

```diff
@@ -313,11 +338,8 @@ def to_package_url(self) -> PackageURL:
             subpath=self._subpath
         )
 
-    def __eq__(self, other: object) -> bool:
-        if isinstance(other, Component):
-            return other.get_purl() == self.get_purl()
-        else:
-            raise NotImplementedError
+    def __eq__(self, other: "Component") -> bool:
```

Comment:
this would cause crashes, if `other` is not implementing `.getPurl()`.
previous implementation was fail safe.

could you describe the reason for this change?